### DISCUSSION TITLE
Interaction between priors and constraints

### DISCUSSION THREAD

**Discussion by AlexanderMouton** [cornellius-gp/gpytorch#2583] [Q&A] [answered]
Hi PyTorch community,

A bit of a technical detail I'm curious about and for which I could not find an answer in the docs/forums: What is the interaction between priors and constraints?

If I place a `Normal(mean=0, std=1)` prior and an `Interval(lower=-2, upper=2)` constraint on a parameter, would the parameter now be "truncated" Normally distributed on [-2, 2]? Or does some transformation magic happen such that the Normal distribution is still defined on (-inf, inf), but operates in the transformed space, with a transformed mean and variance, instead?

I don't see how the latter would be the case, but want to be clear on the details, thanks! :)

**Comment by gpleiss** [comment, marked as answer]
Good question. AFIAK, the prior is applied to the non-constrained parameter, and then the constraints are applied.

**Reply by AlexanderMouton** [reply]
Thanks! Truncated Normal then :)
